iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/snapshot/SnapshotTaker.java

```diff
@@ -206,10 +206,10 @@ private boolean createSnapshot(List<TsFileResource> resources, String snapshotId
         createHardLink(
             new File(snapshotTsFile.getAbsolutePath() + TsFileResource.RESOURCE_SUFFIX),
             new File(tsFile.getAbsolutePath() + TsFileResource.RESOURCE_SUFFIX));
-        if (resource.getModFile().exists()) {
+        if (resource.exclusiveModFileExists()) {
```

Comment:
Also copy the shared mods files if it does not exist in snapshot dir.

Reply:
Done